core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -106,6 +104,12 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
   // Authentication helper used when serving method calls via socket from Python side.
   private lazy val authHelper = new SocketAuthHelper(conf)
 
+  // each python worker gets an equal part of the allocation. the worker pool will grow to the
+  // number of concurrent tasks, which is determined by the number of cores in this executor.
+  private def getWorkerMemoryMb(mem: Option[Long]): Option[Long] = {
+    mem.map(_ / conf.get(EXECUTOR_CORES))
```

Comment:
If so, `EXECUTOR_CORES` should also comes from `ResourceProfile`?

Reply:
yes definitely should, I missed that